### ISSUE TITLE
Compiled files size is too large

### ISSUE THREAD

**Issue by vitormf** [CocoaPods/cocoapods-packager#70]
The framework file being created by the packager is much larger than when I build with XCode.

I've noticed that each compiled file (.o) with the packager is about 2 times the size of the compiled file when building inside XCode.
It's even a little bit larger than when I build in debug mode.

I'm not sure if there is a config file for the packager or if I'm doing something else wrong.
I've installed cocoapods-packager through gem and I'm just using the command "pod package MyProject.podspec"

**Comment by segiddins** [comment]
That's because the packager is building for 3 (2) architectures instead of 1.

**Comment by vitormf** [comment]
Hey Samuel,

Thanks for the fast response.

I've noticed that it's building for 3 architectures (arm64, armv7 and armv7s), while building in xcode it build for 2 (arm64, armv7).
But what I meant is that the intermediary file for one architecture is about double the size of the compiled file in the xcode.

For example, I have a class named DeviceInfo. 
I've analyzed the compiled DeviceInfo.o file.
- Building this class with the XCode it in arm64 it has 17kb and in armv7 it has 13 kb. Combining 30kb.
- Building this class with the XCode in debug mode it has 30kb (arm64) and 29kb (armv7). Combining 59kb.
- Building with the packager it has 31kb (arm64), 26kb (armv7) and 26kb (armv7s). Combining 83kb.

The total size of my library file when building with xcode is 5.7Mb and with the packager is 19.9Mb.

**Comment by segiddins** [comment]
Do you have any dead code stripping options enabled in your Xcode project? What are your optimization settings?

**Comment by vitormf** [comment]
In debug mode: None -O0
In release mode: Fastest, Smallest -Os

**Comment by vitormf** [comment]
Dead code stripping: Yes

**Comment by neonichu** [comment]
You could diff the output of `nm -g` for the two cases and see if anything obvious sticks out.

**Comment by vitormf** [comment]
Hey Boris,
Thanks for your response.
I've run this command in both .o files.
The diff between the two, is that the code generated by the xcode contained one extra symbol "_objc_storeStrong".
Also the numeric value on the left column were different for all symbols that had one (which I suppose it must be the position in memory for that symbol)

**Comment by vitormf** [comment]
Here are all arguments that the XCode passes when compiling that specific class:
https://dl.dropboxusercontent.com/u/19869922/inloco/xcode_build_arguments.txt

I've broken some lines to enable easier reading.
Is there a way for the packager to list the compiling arguments?

**Comment by neonichu** [comment]
A `puts` here <https://github.com/CocoaPods/cocoapods-packager/blob/bfa05c8fb3578bffb8828821cb57a9ae34bdd83d/lib/cocoapods-packager/builder.rb#L174> should show you all output of `xcodebuild`

**Comment by vitormf** [comment]
Boris,

I've running the packager stopping at each step.
Before downloading the code, it creates in a temporary directory a file named Pods.xcodeproj with the target for the Pods and another for my library.

In my original project it had the configuration "Generate Debug Symbols" set to "NO" while in this temporary project it is set to "YES".

Do you have any idea what might be causing it?

**Comment by vitormf** [comment]
When I concat GCC_GENERATE_DEBUGGING_SYMBOLS='NO'  in the compile method return value, it generates the framework with the correct size. 
Though I'm not sure if that would be the correct fix for this issue.

**Comment by vitormf** [comment]
Correction: Not the return value. Concat with the "defines" variable in the compile method.

**Comment by blakewatters** [comment]
I’m not sure if this is really a problem — the debugging symbols are helpful when developers are working with your framework during development and the App Store build process will strip them. The LayerKit uncompressed framework is about 75MB but the total weight to an app store build is ~1.1MB.

**Comment by vitormf** [comment]
That is true. For the library to have debug symbols or not, it doesn't affect the final product as it seems that when exporting for app store or adhoc, the xcode strips them.

In our case, we're choosing to export without the debug symbols, because it makes the download of the library smaller. And since the source code is not provided, it  wouldn't help the developers to debug anyway.

We've choosen to use the cocoapods packager mostly due to the mangling of symbols.
But we also needed to be able to build from local files and strip the debug symbols (as explained above).
In the end, we've decided to rewrite the script, reusing the mangling part.

**Comment by neonichu** [comment]
I think it makes sense to provide an option to optionally strip the build product for the purpose of smaller downloads, if desired.
